components/match2/commons/match_ticker/match_ticker_query.lua

```diff
@@ -0,0 +1,186 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:MatchTicker/Query
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local String = require('Module:StringUtils')
+local Logic = require('Module:Logic')
+local Table = require('Module:Table')
+
+local Condition = require('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local _CURRENT_DATE_STAMP = os.date('%Y-%m-%d %H:%M', os.time(os.date('!*t')))
+
+local _LIMIT_INCREASE = 20
+local _DEFAULT_LIMIT = 20
+local _DEFAULT_QUERY_COLUMNS = {
+	'match2opponents',
+	'winner',
+	'pagename',
+	'tournament',
+	'tickername',
+	'icon',
+	'date',
+	'publishertier',
+	'vod',
+	'stream',
+	'extradata',
+	'parent',
+	'finished',
+	'bestof',
+	'match2id',
+	'icondark',
+}
+local _DEFAULT_ODER = 'date asc, liquipediatier asc, tournament asc'
+
+local MatchTickerQuery = Class.new()
+MatchTickerQuery.maximumLiveHoursOfMatches = 3
+
+local Query = Class.new(
+	function(self)
+		self.queryColumns = _DEFAULT_QUERY_COLUMNS
+		self.orderValue = _DEFAULT_ODER
+		self.limitValue = _DEFAULT_LIMIT + _LIMIT_INCREASE
+	end
+)
+
+function Query:addQueryColumn(queryColumn)
+	table.insert(self.queryColumns, queryColumn)
+	return self
+end
+
+function Query:setOrder(order)
+	self.orderValue = order
+	return self
+end
+
+function Query:setConditions(conditions)
+	self.conditions = conditions
+	return self
+end
+
+function Query:setLimit(limit)
+	-- increase the limit in case we have inelligable matches
```

Comment:
Don't understand this

Reply:
i see 3 kinds of inelligable matches:
1. matches with lpdb bugs that we can not fix (i.e. missing opponent append) (the one where it appends the players to wrong opponents we fix via the according work-around-module for it)
2. for player pages: recent matches that contain a "TBD" or empty opponent
3. for main page mass "TBD"(or empty) vs "TBD"(or empty) from the same page

and since match2 query are slow af i want to keep them minimal hence rather increse the limit a bit and be on the safe side when kicking matches so we still get to the limit value that was entered

 if we do not want that safety we surely can kick the ` + _LIMIT_INCREASE` (plus this comment)

Reply:
Put that into the comment instead :)

Reply:
done